docs/_ext/pharmpy_snippet.py

```diff
@@ -0,0 +1,199 @@
+        self.state.nested_parse(
+            ViewList(
+                string2lines('\n'.join(lines)),
+                source='[SnippetDirective]',
+            ),
+            self.content_offset,
+            node,
+        )
+
+        return node.children
+
+        # for idx, line in enumerate(new_content):
+            # self.content.data.insert(idx, line)
+            # self.content.items.insert(idx, (None, idx))
+
+        # del self.content.data[len(new_content):]
+        # del self.content.items[len(new_content):]
+
+        # node = nodes.container()
+        # self.state.nested_parse(self.content, self.content_offset, node)
+
+class PharmpyExecute(RecursiveDirective):
+    required_arguments = 0
+    optional_arguments = 0
+    final_argument_whitespace = True
+    has_content = True
+
+    option_spec = {
+        "hide-code": directives.flag,
+        "hide-output": directives.flag,
+        "code-below": directives.flag,
+        "linenos": directives.flag,
+        "lineno-start": directives.nonnegative_int,
+        "emphasize-lines": directives.unchanged_required,
+        "raises": csv_option,
+        "stderr": directives.flag,
+    }
+
+    def run(self):
+        # if 'hide-code' in self.options:
+            # raise ValueError('pharmpy-execute does not allow to hide code, use jupyter-execute instead')
+
+        return self._nodes()
+
+    def _nodes(self):
+        lines = self._lines()
+        return self._convert_lines_to_nodes(lines)
+
+    def _lines(self):
+        return [
+            f'.. container:: pharmpy-snippet{"" if "hide-output" in self.options else " with-output"}',
+            '',
+            *_indent(3, self._input_output_lines())
+        ]
+
+    def _input_output_lines(self):
+        if 'hide-code' in self.options:
+            return self._output()
+
+        if 'code-below' in self.options:
+            return [
+                *self._output(),
+                '',
+                *self._input(),
+            ]
+
+        return [
+            *self._input(),
+            '',
+            *self._output(),
+        ]
+
+    def _input(self):
+        return [
+            '.. tabs::',
+            *_indent(3, [
+                '',
+                '.. code-tab:: py',
+                *_indent(3, self._code_option_lines()),
+                '',
+                *_indent(3, self.content),
+                '',
+                '.. code-tab:: r R',
+                *_indent(3, self._code_option_lines()),
+                '',
+                *_indent(3, transpile_py_to_r(self.content)),
+            ]),
+        ]
+
+    def _code_option_lines(self):
+        if 'emphasize_lines' in self.options:
+            yield f':emphasize_lines:{self.options.get("emphasize_lines")}'
+        if "linenos" in self.options:
+            yield ':linenos:'
+        if "lineno-start" in self.options:
+            yield f':lineno-start:{self.options.get("lineno-start")}'
```

Comment:
`emphasize_lines` and `lineno-start` may not work well with tranpsiled code, but that could be fixed later.